samples/javascript_nodejs/56.teams-file-upload/services/fileService.js

```diff
@@ -0,0 +1,52 @@
+const fs = require('fs');
+const util = require('util');
+const axios = require('axios');
+const readdir = util.promisify(fs.readdir);
+
+// Generates File Name with Sequence
+const GeneFileName = async (fileDir) => {
+    const filenameConst = 'UserAttachment';
+    const files = await readdir(fileDir);
+    const filteredFiles = files.filter(f => f.includes(filenameConst)).map(f => parseInt(f.split(filenameConst)[1].split('.')[0]));
+    let maxSeq = 0;
+    if (filteredFiles.length > 0) {
+        maxSeq = Math.max.apply(Math, filteredFiles);
+    } else {
+        maxSeq = 0;
+    }
+    const filename = `${ filenameConst }${ maxSeq + 1 }.png`;
+    return filename;
+};
+
+// Download and Save Streams into File
+const WriteFile = async (contentUrl, config, filePath) => {
+    return new Promise((resolve) => {
+        axios.get(contentUrl, config).then(response => {
+            const stream = response.data.pipe(fs.createWriteStream(filePath));
+            stream.on('finish', () => {
+                resolve();
+            });
+        });
+    });
+};
+// const WriteFile = async (contentUrl, config, filePath) => {
+//     return new Promise(async (resolve) => {
+//         const response = await axios.get(contentUrl, config);
+//         const stream = response.data.pipe(fs.createWriteStream(filePath));
+//         stream.on('finish', () => {
+//             resolve();
+//         });
+//     });
+// };
```

Comment:
Let's remove this commented code, then rename these functions to be `camelCase` (lowercase the first letter), then I think we're good. In VS Code, you can highlight the name, press F2, then rename and it should rename throughout all of your code--good to double-check it, though, where used.

I'm not super familiar with using `axios` with streams and I _believe_ your code for `WriteFile()` will work. I recommend testing it, specifically, if you haven't already. I'm not sure if you need to add `{ responseType: 'stream' }`?